sink/datadog.go

```diff
@@ -141,13 +160,70 @@ func NewDatadog(monitorId string, opts, tags map[string]string, httpClient *http
 
 		c := datadog.NewConfiguration()
 		c.HTTPClient = httpClient
+		c.Compress = d.compress
 		metricsApi := datadogV2.NewMetricsApi(datadog.NewAPIClient(c))
 		d.metricsApi = metricsApi
 	}
 
 	return d, nil
 }
 
+func (s *Datadog) sendApi(ddCtx context.Context, dp []datadogV2.MetricSeries) error {
+	// Send metrics to Datadog. The Datadog client handles everything; we just pass
+	// it data points.
+	payload := datadogV2.MetricPayload{
+		Series: dp,
+	}
+	_, r, err := s.metricsApi.SubmitMetrics(ddCtx, payload, *datadogV2.NewSubmitMetricsOptionalParameters())
+	// Check to see if the request was too large
+	if err != nil {
+		if r.StatusCode == http.StatusRequestEntityTooLarge {
+			oldMaxMetricsPerRequest := s.maxMetricsPerRequest
+
+			// Sample one of the metrics and try caculating a new maxMetricsPerRequest
+			if pErr := s.updateMaxMetricsPerRequest(dp[0], len(dp)); pErr != nil {
```

Comment:
Don't use `pErr` here in a block scope but then again later in another (outer) block scope. They have same special name but are different, which contradicts the normal expectation that any specifically-named var is the same var wherever the specific name occurs. (This is not true for nonspecifically named vars like the ubiquitous `err` in Go, and other classics like `i` in loops, etc.)